zenoh/src/net/routing/interceptor/downsampling.rs

```diff
@@ -0,0 +1,160 @@
+            }
+        }
+
+        let timestamp = std::time::Instant::now();
+        let mut latest_message_timestamp = self.latest_message_timestamp.lock().unwrap();
+
+        if timestamp - *latest_message_timestamp >= self.threshold {
+            *latest_message_timestamp = timestamp;
+            log::debug!("Interceptor: Passed threshold, passing.");
+            Some(ctx)
+        } else {
+            log::debug!("Interceptor: Skipped due to threshold.");
+            None
+        }
+    }
+}
+
+impl EgressMsgDownsamplerRatelimit {
+    pub fn new(conf: DownsamplingItemConf) -> Self {
+        if let Some(threshold_ms) = conf.threshold_ms {
+            let threshold = std::time::Duration::from_millis(threshold_ms);
+            Self {
+                keyexprs: conf.keyexprs,
+                threshold,
+                // TODO (sashacmc): I need just := 0, but how???
+                latest_message_timestamp: Arc::new(Mutex::new(
+                    std::time::Instant::now() - threshold,
+                )),
+            }
+        } else {
+            // TODO (sashacmc): how correctly process an error?
+            panic!("Rate limit downsampler shoud have a threshold_ms parameter");
+        }
+    }
+}
+
+pub struct DownsamplerInterceptor {
+    conf: DownsamplingItemConf,
+}
+
+impl DownsamplerInterceptor {
+    pub fn new(conf: DownsamplingItemConf) -> Self {
+        log::debug!("DownsamplerInterceptor enabled: {:?}", conf);
+        Self { conf }
+    }
+}
+
+impl InterceptorFactoryTrait for DownsamplerInterceptor {
+    fn new_transport_unicast(
+        &self,
+        transport: &TransportUnicast,
+    ) -> (Option<IngressInterceptor>, Option<EgressInterceptor>) {
+        log::debug!("New downsampler transport unicast {:?}", transport);
+        if let Some(interfaces) = &self.conf.interfaces {
+            log::debug!(
+                "New downsampler transport unicast config interfaces: {:?}",
+                interfaces
+            );
+            if let Ok(links) = transport.get_links() {
+                for link in links {
+                    log::debug!(
+                        "New downsampler transport unicast link interfaces: {:?}",
+                        link.interfaces
+                    );
+                    if !link.interfaces.iter().any(|x| interfaces.contains(x)) {
+                        return (None, None);
+                    }
+                }
+            }
+        };
+
+        let strategy = self
+            .conf
+            .strategy
+            .as_ref()
+            .map_or_else(|| RATELIMIT_STRATEGY.to_string(), |s| s.clone());
+
+        if strategy == RATELIMIT_STRATEGY {
+            (
+                None,
+                Some(Box::new(EgressMsgDownsamplerRatelimit::new(
+                    self.conf.clone(),
+                ))),
+            )
+        } else {
+            panic!("Unsupported downsampling strategy: {}", strategy)
```

Comment:
I think we could log in debug and return `(None, None)`.

Reply:
Debug? Maybe better error or at least warning? It seems to me that this is how we will "hide" the configuration error.